File: iridium-flask-client/model/identity_response.py
# python response object containing id, username, profile_response
class identity_response(object):
    # python function returning object containing id, username, profile_response
    def __init__(self, id, username, profile_response):
        # python set id for id
        self.id = id
        # python set username for username
        self.username = username
        # python set profile_response for profile_response
        self.profile_response = profile_response

    # python function returning json object containing id, username, profile_response
    def to_json(self):
        # python set payload for http client
        payload = {'id': self.id, 'username': self.username, 'profile_response': self.profile_response}
        return payload
    
    # python function returning string containing id, username, profile_response
    def __str__(self):
        # python set payload for http client
        payload = {'id': self.id, 'username': self.username, 'profile_response': self.profile_response}
        return str(payload)
    

File: iridium-java-client/src/main/java/software/iridium/api/authentication/domain/IdentityResponse.java
/*
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */
package software.iridium.api.authentication.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class IdentityResponse implements Serializable {

  public static final String MEDIA_TYPE = "application/vnd.iridium.id.identity-response.1+json";

  private static final long serialVersionUID = 7221668651862331716L;

  private String id;

  private String username;

  private ProfileResponse profile;

  @JsonIgnore private String redirectUri;

  private Set<String> roles = new HashSet<>();

  private List<String> tenantIds = new ArrayList<>();

  public String getUsername() {
    return username;
  }

  public void setUsername(final String username) {
    this.username = username;
  }

  public Set<String> getRoles() {
    return roles;
  }

  public void setRoles(final Set<String> roles) {
    this.roles = roles;
  }

  public String getId() {
    return id;
  }

  public void setId(final String id) {
    this.id = id;
  }

  public ProfileResponse getProfile() {
    return profile;
  }

  public void setProfile(final ProfileResponse profile) {
    this.profile = profile;
  }

  public List<String> getTenantIds() {
    return tenantIds;
  }

  public void setTenantIds(final List<String> tenantIds) {
    this.tenantIds = tenantIds;
  }

  public String getRedirectUri() {
    return redirectUri;
  }

  public void setRedirectUri(final String redirectUri) {
    this.redirectUri = redirectUri;
  }

  @Override
  public String toString() {
    return "IdentityResponse{"
        + "id='"
        + id
        + '\''
        + ", username='"
        + username
        + '\''
        + ", profile="
        + profile
        + '\''
        + ", roles="
        + roles
        + ", tenantIds="
        + tenantIds
        + '}';
  }
}
